### PR TITLE
Add networking support for Go applications

### PR DESCRIPTION
**Pull request by razvand**
Add networking support for Go applications. Demonstrate using `server.go`, a simple HTTP-like server written in Go.

To build `server.go`, first uncomment the `server.go` line in `Makefile.uk`, and comment out the `helloworld.go` line.

Then, build the Unikraft image:
    
```console
kraft build --kraftfile kraft.cloud.yaml
```
    
Run the Unikraft Go server image:
    
```console
sudo kraft net rm kraft0
sudo kraft net create -n 172.44.0.1/24 kraft0
sudo kraft run --kraftfile kraft.cloud.yaml --plat firecracker --network bridge:kraft0 -M 256M -a netdev.ipv4_addr=172.44.0.2 -a netdev.ipv4_gw_addr=172.44.0.1 -a netdev.ipv4_subnet_mask=255.255.255.0 -- "server"
```

### CODE REVIEW

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Review by StefanJum — CHANGES_REQUESTED** [review]
@razvand you left an unclosed bracket in the commit message `(and comment out the helloworld.go line)`.

Besides that, what is the `fs0` initrd directory? If it's not needed remove it from the run command in the commit message.

File: server.go
Comment (jschlumpp) [inline server.go:1]: It might be a good idea to run `go fmt` on the file. The file isn't formatted according to the official Go style.

**Review by StefanJum — APPROVED** [review]
Works fine, I'll add the tag after you format the server file, as @mschlumpp suggested.

**Comment by razvand** [comment]
@StefanJum , @mschlumpp , fixed.

**Review by StefanJum — APPROVED** [review]
All good, thanks.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by jschlumpp — APPROVED** [review]
Edit: I haven't noticed that I was assigned ^^

Reviewed-by: Marco Schlumpp <marco@unikraft.io>
Approved-by: Marco Schlumpp <marco@unikraft.io>

**Pull request closed without merging**